src/controller/CHIPDiscovery.cpp

```diff
@@ -0,0 +1,336 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <stdarg.h>
+
+#include <cstdint>
+#include <string>
+
+#include <controller/CHIPDiscovery.h>
+#include <lib/mdns/minimal/core/Constants.h>
+
+namespace chip {
+namespace Controller {
+
+namespace {
+std::string ToString(const mdns::Minimal::BytesRange & bytes)
+{
+    std::string str;
```

Comment:
Could you use the constructor:

```cpp
return std::string(reinterpret_cast<const char *>(bytes.Start()), bytes.Size());
```

?

Reply:
Are we ok to use std::string ? we generally seem to try to avoid them because embedded and trying to control heap.

Reply:
Are there going to be embedded controllers? If so, I'm fine to use C style.

Reply:
removed strings in the new commit.